skglm/datafits/single_task.py

```diff
@@ -774,3 +773,114 @@ def _get_indptr(self, vals, indices):
         indptr.append(n_indices)
 
         return np.asarray(indptr, dtype=np.int64)
+
+
+class WeightedQuadratic(BaseDatafit):
```

Comment:
I think this could be moved closer to the Quadratic class (at the top of the file)